src/authutil.ts

```diff
@@ -0,0 +1,139 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as xmlbuilder from 'xmlbuilder';
+import * as xmlParser from 'fast-xml-parser';
+import {booleanLiteral} from '@babel/types';
+
+export function configAuthentication(
+  feedUrl: string,
+  existingFileLocation: string = ''
+) {
+  const existingNuGetConfig: string = path.resolve(
+    process.env['RUNNER_TEMP'] || process.cwd(),
+    existingFileLocation == '' ? 'nuget.config' : existingFileLocation
+  );
+
+  const tempNuGetConfig: string = path.resolve(
+    process.env['RUNNER_TEMP'] || process.cwd(),
+    '../',
+    'nuget.config'
+  );
+
+  writeFeedToFile(feedUrl, existingNuGetConfig, tempNuGetConfig);
+}
+
+function writeFeedToFile(
+  feedUrl: string,
+  existingFileLocation: string,
+  tempFileLocation: string
+) {
+  console.log(
+    `dotnet-auth: Finding any source references in ${existingFileLocation}, writing a new temporary configuration file with credentials to ${tempFileLocation}`
+  );
+  let xml: xmlbuilder.XMLElement;
+  let gprKeys: string[] = [];
+  let owner: string = core.getInput('owner');
+  if (!owner && feedUrl.indexOf('nuget.pkg.github.com') > -1) {
+    owner = github.context.repo.owner;
+  }
+  let sourceUrl: string = 'https://nuget.pkg.github.com/' + owner;

```

Comment:
This should be `https://nuget.pkg.github.com/{owner}/index.json`

Reply:
🤦‍♂ thanks. Somehow that got lost.